tests/stdlib/tsetutils.nim

```diff
@@ -0,0 +1,8 @@
+discard """
+  targets: "c js"
+"""
+import std/setutils
+
+doAssert "abcbb".toSet == {'a','b','c'}
+doAssert toSet([10u8,12,13]) == {10u8, 12, 13}
+doAssert toSet(0u16..30) == {0u16..30}
```

Comment:
add a test with `distinct`
```nim
type A = distinct char
doAssert ['x'.A].toSet == {'x'.A}
```

this should work but will fail currently in your PR because of `static: doAssert E is SetElement` 

you'd at least need `static: doAssert distinctBase(E) is SetElement`, but as I suggested above might as well simply skip this redundant check since compiler will do that for you anyways when you call `var result: set[E]`